src/Idris/Parser.hs

```diff
@@ -433,20 +433,20 @@ syntaxRule syn
             reservedHL "syntax"
             return sty)
          syms <- some syntaxSym
-         when (all isExpr syms) $ P.unexpected "missing keywords in syntax rule"
+         when (all isExpr syms) $ P.unexpected . P.Label $ 'm' :| "issing keywords in syntax rule"
```

Comment:
What does `:|` do?

Reply:
It's cons for `NonEmpty`, which is what `P.Label` takes for the label name.  Elsewhere I used `NonEmpty.fromList`, which throws if an error if given an empty list.  I don't really like either way so... I used both?

Do you have a preference?

Reply:
NonEmpty.fromList seems OK since we statically have a string 😄